miner/tx_pool_w_accounts.go

```diff
@@ -0,0 +1,111 @@
+package miner
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/pending_txs"
+	"github.com/spacemeshos/sha256-simd"
+	"sort"
+	"sync"
+)
+
+type innerPool interface {
+	Get(id types.TransactionId) (types.Transaction, error)
+	GetAllItems() []types.Transaction
+	Put(id types.TransactionId, item *types.Transaction)
+	Invalidate(id types.TransactionId)
+}
+
+type TxPoolWithAccounts struct {
+	innerPool
+	accounts      map[types.Address]*pending_txs.AccountPendingTxs
+	accountsMutex sync.RWMutex
+}
+
+func NewTxPoolWithAccounts() *TxPoolWithAccounts {
+	return &TxPoolWithAccounts{
+		innerPool: NewTxMemPool(),
+		accounts:  make(map[types.Address]*pending_txs.AccountPendingTxs),
+	}
+}
+
+func (t *TxPoolWithAccounts) GetRandomTxs(numOfTxs int, seed []byte) []types.Transaction {
+	txs := t.innerPool.GetAllItems()
+	if len(txs) <= numOfTxs {
+		return txs
+	}
+
+	sort.Slice(txs, func(i, j int) bool {
+		id1 := txs[i].Id()
+		id2 := txs[j].Id()
+		return bytes.Compare(id1[:], id2[:]) < 0
+	})
+	var ret []types.Transaction
+	for idx := range getRandIdxs(numOfTxs, len(txs), seed) {
```

Comment:
Why do you need `getRandIdxs`? Why not just randomly pick a number * `len(txs)` iterations?

Reply:
Let's discuss f2f.